src/Symfony/Component/Security/Http/Tests/Firewall/ContextListenerTest.php

```diff
@@ -47,19 +47,6 @@ public function testItRequiresContextKey()
         );
     }
 
-    /**
-     * @expectedException \InvalidArgumentException
-     * @expectedExceptionMessage User provider "stdClass" must implement "Symfony\Component\Security\Core\User\UserProviderInterface
```

Comment:
we should still have a test covering this case. You have not removed the exception entirely. You just moved it later in the usage.